### ISSUE TITLE
Для чего нужно декларировать elem?

### ISSUE THREAD

**Issue by mikvys** [bem/bem-forum-content-ru#439]
Привет ребят. Я не могу понять для чего нужен elem в bemjson. Ведь можно всё сделать через блоки, и при необходимости использовать mix для того, чтобы доопределить что нибудь? Или я скорей всего чего то не догоняю) Подскажите пожалуйста?

**Comment by tadatuta** [comment]
@mikvys Самая простая аналогия для элементов — это Shadow DOM в спеке Web Components. Те части блоков, которые не должны «торчать наружу», а являются неотъемлемой частью блока и без него не имеют смысла.

**Comment by mikvys** [comment]
Если у меня есть блок search, то текстовое поле - это ведь его неотъемлемая часть? В то же время оно будет "торчать наружу" так как текстовое поле может несколько раз использоваться на странице. Значит мы помещаем блок внутрь блока, и потом можем доопределить блок текстовое поле в блоке search используя mix. Тогда выходит, что мы помещаем элемент внутрь блока только тогда, когда это что то уникальное, то, что не может использоваться больше нигде?

**Comment by tadatuta** [comment]
Нет, просто поисковое поле будет миксом блока `input` и элементом блока `form`:
```js
{
    block: 'form',
    content: {
        block: 'input',
        mix: { block: 'form', elem: 'input' }
    }
}
```

И автоматически получается логичное разделение: весь код, который действительно универсальный, окажется в `input`, а конкретика про какие-нибудь отступы, нужные именно в данной форме — в `form__input`.

**Comment by mikvys** [comment]
Спасибо, кажется разобрался :)
